discovery-provider/integration_tests/tasks/test_index_playlists.py

```diff
@@ -333,3 +338,161 @@ def test_index_playlist(app):
         )
         assert playlist_record.is_delete == True
+
+
+def test_playlist_indexing_skip_tx(app, mocker):
+    """Tests that playlists skip cursed txs without throwing an error and are able to process other tx in block"""
+    with app.app_context():
+        db = get_db()
+        ipfs_client = IPFSClient({})
+        web3 = Web3()
+        challenge_event_bus = setup_challenge_bus()
+        update_task = UpdateTask(ipfs_client, web3, challenge_event_bus)
+
+    class TestPlaylistTransaction:
+        pass
+
+    blessed_tx_hash = (
+        "0x34004dfaf5bb7cf9998eaf387b877d72d198c6508608e309df3f89e57def4db3"
+    )
+    blessed_tx = TestPlaylistTransaction()
+    blessed_tx.transactionHash = update_task.web3.toBytes(hexstr=blessed_tx_hash)
+    cursed_tx_hash = (
+        "0x5fe51d735309d3044ae30055ad29101018a1a399066f6c53ea23800225e3a3be"
+    )
+    cursed_tx = TestPlaylistTransaction()
+    cursed_tx.transactionHash = update_task.web3.toBytes(hexstr=cursed_tx_hash)
+    test_block_number = 25278765
+    test_block_timestamp = 1
+    test_block_hash = update_task.web3.toHex(block_hash)
+    test_playlist_factory_txs = [cursed_tx, blessed_tx]
+    test_timestamp = datetime.utcfromtimestamp(test_block_timestamp)
+    blessed_playlist_record = Playlist(
+        blockhash=test_block_hash,
+        blocknumber=test_block_number,
+        txhash=blessed_tx_hash,
+        playlist_id=91232,
+        is_album=False,
+        is_private=False,
+        playlist_name="test",
+        playlist_contents={},
+        playlist_image_multihash=None,
+        playlist_image_sizes_multihash=None,
+        description="testing!",
+        upc=None,
+        is_current=True,
+        is_delete=True,
+        last_added_to=None,
+        updated_at=test_timestamp,
+        created_at=test_timestamp,
+        playlist_owner_id=1,
+    )
+    cursed_playlist_record = Playlist(
+        blockhash=test_block_hash,
+        blocknumber=test_block_number,
+        txhash=cursed_tx_hash,
+        playlist_id=91238,
+        is_album=None,
+        is_private=None,
+        playlist_name=None,
+        playlist_image_multihash=None,
+        playlist_image_sizes_multihash=None,
+        description=None,
+        upc=None,
+        is_current=True,
+        is_delete=True,
+        last_added_to=None,
+        updated_at=test_timestamp,
+        created_at=None,
+    )
+
+    mocker.patch(
+        "src.tasks.playlists.lookup_playlist_record",
+        side_effect=[cursed_playlist_record, blessed_playlist_record],
+        autospec=True,
+    )
+    mocker.patch(
+        "src.tasks.playlists.get_playlist_events_tx",
+        side_effect=[
+            [],  # no playlist created events
+            [
+                {
+                    "args": {
+                        "_playlistId": cursed_playlist_record.playlist_id,
+                    }
+                },
+            ],  # playlist deleted event
+            [],
+            [],
```

Comment:
they're here because the mock returns these in sequence each time the fn is called. this fn gets called once per event type per playlist tx, so for 2 txs total this makes 20 calls 😓

Reply:
resolved. this is because indexing iterates through each playlist event type and calls get_playlist_events_tx for that